api/src/user/user.handlers.ts

```diff
@@ -247,3 +250,167 @@ export const getUserWorksheet = async (
+          eq(worksheetMetadata.season, season),
+        ),
+      )
+      .orderBy(worksheetMetadata.worksheetNumber);
+
+    const worksheetNumbers = worksheetNumbersRes.map(
+      (row) => row.worksheetNumber,
+    );
+
+    const firstAvailableWsNumber = getFirstAvailableWsNumber(worksheetNumbers);
+
+    await db.insert(worksheetMetadata).values({
+      netId,
+      season,
+      worksheetNumber: firstAvailableWsNumber,
+      worksheetName: 'New Worksheet',
+    });
+  } else if (action === 'delete') {
+    const { worksheetNumber } = bodyParseRes.data;
+    const [existingWorksheet] = await db
+      .select()
+      .from(worksheetMetadata)
+      .where(
+        and(
+          eq(worksheetMetadata.netId, netId),
+          eq(worksheetMetadata.season, season),
+          eq(worksheetMetadata.worksheetNumber, worksheetNumber),
+        ),
+      );
+
+    if (!existingWorksheet) {
+      res.status(400).json({ error: 'WORKSHEET_NOT_FOUND' });
+      return;
+    }
+
+    winston.info(`Deleting worksheet ${worksheetNumber} for user ${netId}`);
+    await db
+      .delete(worksheetMetadata)
+      .where(
+        and(
+          eq(worksheetMetadata.netId, netId),
+          eq(worksheetMetadata.season, season),
+          eq(worksheetMetadata.worksheetNumber, worksheetNumber),
+        ),
+      );
+  } else {
+    const { worksheetNumber, worksheetName } = bodyParseRes.data;
+    const [existingWorksheet] = await db
+      .select()
+      .from(worksheetMetadata)
+      .where(
+        and(
+          eq(worksheetMetadata.netId, netId),
+          eq(worksheetMetadata.season, season),
+          eq(worksheetMetadata.worksheetNumber, worksheetNumber),
+        ),
+      );
+
+    if (!existingWorksheet) {
+      res.status(400).json({ error: 'WORKSHEET_NOT_FOUND' });
+      return;
+    }
+    winston.info(
+      `Renaming worksheet ${worksheetNumber} for user ${netId} to "${worksheetName}"`,
+    );
+    await db
+      .update(worksheetMetadata)
+      .set({ worksheetName })
+      .where(
+        and(
+          eq(worksheetMetadata.netId, netId),
+          eq(worksheetMetadata.season, season),
+          eq(worksheetMetadata.worksheetNumber, worksheetNumber),
+        ),
+      );
+  }
+
+  res.sendStatus(200);
+};
+
+export const getUserWorksheetMetadata = async (
+  req: express.Request,
+  res: express.Response,
+): Promise<void> => {
+  winston.info(`Fetching worksheet names for user`);
+
+  const { netId } = req.user!;
+
+  const allWorksheetMetadata = await db
+    .select({
+      season: worksheetMetadata.season,
+      worksheetNumber: worksheetMetadata.worksheetNumber,
+      worksheetName: worksheetMetadata.worksheetName,
+    })
+    .from(worksheetMetadata)
+    .where(eq(worksheetMetadata.netId, netId));
+
+  winston.info(
+    `Retrieved ${allWorksheetMetadata.length} worksheets for user ${netId}`,
+  );
+
+  const worksheetMap: {
+    [season: string]: { [worksheetNumber: number]: string } | undefined;
```

Comment:
This doesn't follow your documented API shape—the API you documented is `{ [worksheetNumber: number]: { worksheetName: string } }` which IMO is much better